uwblueprint/headsupguys
# Mood tracker frontend

## Notion ticket link

No ticket yet



## Implementation description
* Added frontend for mood tracker
    * Added buttons on home page
    * Added modal
    * Connected to backend



## Steps to test
1. Make sure you are signed out and go to http://localhost:3000/
2. Observe no "Track mood" button and no chart icon button
3. Sign in
4. Observe "Track mood" button (below video) and chart icon button (top right, next to header):
![Screen Shot 2022-04-17 at 3 18 11 PM](https://user-images.githubusercontent.com/49915445/163729150-7c06f8bc-ad06-4106-b559-b7ea69179629.png)
5. Click on both buttons and confirm they open up this modal (note that the month and year should be the current ones):
![Screen Shot 2022-04-17 at 3 19 09 PM](https://user-images.githubusercontent.com/49915445/163729174-b9a6b094-8233-48dc-91ce-340051389035.png)
6. Observe the 5 options and pick one:
![Screen Shot 2022-04-17 at 3 20 26 PM](https://user-images.githubusercontent.com/49915445/163729216-d8a8e1e4-2221-4b69-8596-b87d0f4bd6fa.png)
7. Bars and select should update:
![Screen Shot 2022-04-17 at 3 21 04 PM](https://user-images.githubusercontent.com/49915445/163729232-0c1532ba-2149-457c-b6c7-067a29e78323.png)
8. Close the modal and observe that the "Track mood" button and section are gone (the section should disappear after a mood is selected for the day)
9. Reopen the modal by clicking on the chart icon button to the right of the header
10. Confirm that the previous selection is still selected and the bar is still there
11. Click on the arrows to toggle months and confirm that: the dates in the bar charts change accordingly, you cannot view bars or months beyond the current date, the bar that was filled persists (i.e. the bar from step 7 should stay)
12. Try changing the current option and confirm that the change is reflected in the bars
13. Test responsiveness of modal and buttons for mobile and desktop
15. If possible, come back another day or change your system clock to repeat the steps for another day



## What should reviewers focus on?
* Does the UI look ok?
* Does it work properly?


## Checklist
